lib/proxy_api/external_ipam.rb

```diff
@@ -128,146 +132,145 @@ def get_groups
+    #   Response if group doesn't exist:
+    #     {"code": 404, "message": "Not found"}
+    #   Response if groups are not supported:
+    #     {"code": 501, "error": "Groups are not supported"}
     #   Response if can't connect to External IPAM server
     #     {"code": 500, "error": "Unable to connect to External IPAM server"}
-    def get_group(group_name)
-      raise "group_name must be provided" if group_name.blank?
-      parse get("/groups/#{URI.escape(group_name)}")
+    def get_group(group)
+      raise "group must be provided" if group.blank?
+      parse get("/groups/#{URI.escape(group)}")
     rescue => e
-      raise ProxyException.new(url, e, N_("Unable to obtain group '#{group_name}' from External IPAM."))
+      raise ProxyException.new(url, e, N_("Unable to obtain group '#{group}' from External IPAM."))
     end
 
-    # Get a list of subnets for the given External IPAM section.
+    # Get a list of subnets for the given External IPAM group.
     #
-    # Params:  1. section_name: The name of the External IPAM section
+    # Params:  1. group:  The name of the External IPAM group containing the subnet.
     #
-    # Returns: An array of subnets on success, or a hash with a "message" key
-    #          containing error on failure.
-
-    # Examples:
-    #   Response if success: {"code": 200, "success": true, "data": [
-    #     {"id":"9","subnet":"100.10.10.0","mask":"24","sectionId":"5","description":"Test Subnet 1"},
-    #     {"id":"10","subnet":"100.20.20.0","mask":"24","sectionId":"5","description":"Test Subnet 2"}
-    #   ],"time":0.018}
+    # Returns: An array of subnets on success, or a hash with an "error" key on failure.
+    #
+    # Responses from Proxy plugin:
+    #   Response if success: {"code": 200, "data": [
+    #     {subnet":"100.10.10.0","mask":"24","description":"Test Subnet 1"},
+    #     {subnet":"100.20.20.0","mask":"24","description":"Test Subnet 2"}
+    #   ]}
     #   Response if no subnets exist in section.
-    #     {"code": 200, "success": true, "data": [], "time": 0.007}
+    #     {"code": 200, "data": []}
     #   Response if section not found:
-    #     {"code": 404, "error": "Section not found in External IPAM."}
+    #     {"code": 404, "error": "Group not found in External IPAM"}
     #   Response if can't connect to External IPAM server
-    #     {"code": 500, "error": "Unable to connect to External IPAM server"}
-    def get_subnets_by_group(group_name)
-      raise "group_name must be provided" if group_name.blank?
-      group = URI.unescape(group_name)
-      parse get("/groups/#{URI.escape(group)}/subnets")
+    #     {"code": 500, "error": "Unable to connect to External IPAM"}
+    def get_subnets_by_group(group)
+      raise "group must be provided" if group.blank?
+      unescaped_group = URI.unescape(group)
+      parse get("/groups/#{URI.escape(unescaped_group)}/subnets")
```

Comment:
What's the point of unescape -> escape? I see you use unescaped version in the exception, I assume that "group" argument is a simple string which can be easily printed out.

Reply:
The issue here is that the `foreman_ipam` plugin also uses this method, and needs to pass the group name escaped, which means it would get escaped twice. The unescape -> escape ensures that group is only escaped once before the API call

Reply:
I am not following. Do you mean foreman_ipam plugin? You only need to call escape once when putting something into an URL. `URI.escape(URI.unescape(group)) == group` so you are not getting anything.

Reply:
The foreman_ipam plugin calls the proxy endpoint to get the IPAM subnets. It doesn't hit IPAM directly. So I have to escape the group before calling the proxy endpoint, and since the proxy endpoint is already escaping, it will get escaped twice

Reply:
I see, you escape it in the plugin:

    @subnets = Phpipam.proxy.get_subnets_by_group(URI.escape(params[:group]))

Well, then don't do that, params hash has all the values unescaped by Rails automatically so it's plain string. This will create this method much easier and very similar to `delete_ip_from_subnet` one down below:

    @subnets = Phpipam.proxy.get_subnets_by_group(params[:group])